### PR TITLE
Model technical metadata rather than use a formless datastream

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: 0001-technical-metadata-model.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Currently we create technical metadata by running JHOVE on every binary.  For XML
+binaries this is very slow and for an object with a lot of binaries (like a book)
+this can block the robot pipeline for everyone.  JHOVE has been an ongoing maintenance
+problem.  We store all JHOVE output which can be excessively verbose and has caused
+operation problems with memory consumption.  We believe that only a small subset of the
+JHOVE output will ever be useful to SDR consumers.  Note: SMPL content ships with
+it's own metadata typically from MediaInfo rather than JHOVE.
+
+Using a data model we give us greater flexibility in how we choose to persist
```
Comment (aaron-collier) [inline 0001-technical-metadata-model.md]: s/we give us/will give us/?

**Comment by peetucket** [comment]
Agree with this.  Only question is that if JHOVE is kept, will there still be performance problems even with mapping to a data model?  In other words, just getting the data from JHOVE is slow (e.g. in the big XML file examples) even before we store it, right?

**Comment by mjgiarlo** [comment]
@peetucket :speech_balloon: 
> Agree with this. Only question is that if JHOVE is kept, will there still be performance problems even with mapping to a data model? In other words, just getting the data from JHOVE is slow (e.g. in the big XML file examples) even before we store it, right?

My take on this was that moving from the output-from-tool-as-datastream model to the more selective property model would help us move away from JHOVE because we will then depend on certain properties being present (*however* they are made present) vs on what form the output takes.

File: 0001-technical-metadata-model.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Context and Problem Statement
+
+Currently we create technical metadata by running JHOVE on every binary.  For XML
+binaries this is very slow and for an object with a lot of binaries (like a book)
+this can block the robot pipeline for everyone.  JHOVE has been an ongoing maintenance
+problem.  We store all JHOVE output which can be excessively verbose and has caused
+operation problems with memory consumption.  We believe that only a small subset of the
+JHOVE output will ever be useful to SDR consumers.  Note: SMPL content ships with
+it's own metadata typically from MediaInfo rather than JHOVE.
```
Comment (mjgiarlo) [inline 0001-technical-metadata-model.md]: `s/it's/its/`

File: 0001-technical-metadata-model.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Using a data model will give us greater flexibility in how we choose to persist
+this data.  The data model will show which fields we believe to
+be important and will allow us to discard that which we do not believe to be important.
```
Comment (mjgiarlo) [inline 0001-technical-metadata-model.md]: And, ultimately, will allow us to decouple our approach from any particular tool like JHOVE.

File: 0001-technical-metadata-model.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Because the data is modeled it is more readily actionable and analyzable, which will
+give value to the data that we preserve.
+
+## Decision Drivers <!-- optional -->
+
+This is particularly important because:
+
+* Google books will clog the robots (because existing process is slow)
+* We want to move off Fedora 3, so having data models allow us to swap persistence layers.
```
Comment (mjgiarlo) [inline 0001-technical-metadata-model.md]: They certainly contribute to that goal!

File: 0001-technical-metadata-model.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+* Certain pathological cases (books with > 700 pages) cause existing robots to run out of memory.
```
Comment (mjgiarlo) [inline 0001-technical-metadata-model.md]: ... and these pathological cases are not particularly hard to find in the Google Books data set, I take it?

Comment (aaron-collier) [reply 0001-technical-metadata-model.md]: Uh... Maybe... we can use some of the symphony data to be aware of what we KNOW has a large page count, but that data is incomplete. Additionally, in symphony, bound series seem to have the same page count, which can be like 20000 pages... but the individual volumes have just a subset of those, so we really don't know which ones may be big.

Comment (mjgiarlo) [reply 0001-technical-metadata-model.md]: Here's what I was getting at: what do we mean by "pathological" here? Do we mean to imply that works like these a corner case? Or that that we'll see them regularly enough that we ought to account for them as more than a corner case?

Comment (aaron-collier) [reply 0001-technical-metadata-model.md]: Gotcha. Hmm..

File: 0001-technical-metadata-model.md (hunk 4 above)
Comment (ndushay) [inline 0001-technical-metadata-model.md]: s/allow/allows/

File: 0001-technical-metadata-model.md (hunk 5 above)
Comment (ndushay) [reply 0001-technical-metadata-model.md]: good point, Mike.  Would be good to give a sense of order of magnitude (10s?  100s?  10,000s?)

File: 0001-technical-metadata-model.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+* Having data models is prerequisite for developing APIs
+* SDR evolution meetings emphasized the need to move to well defined data models
+
+## Considered Options
+
+* Get rid of technical metadata entirely.  This might conflict with requirements for being a trusted digital repository
+* Pursue technical fixes to JHOVE and the technical metadata robot
+* Replace JHOVE with other characterization tools. Crosswalk data to a data model.
+  * Use the [Princeton data model](https://github.com/pulibrary/figgy/blob/master/app/resources/nested_resources/file_metadata.rb#L4-L35) as a starting point.
+  * We anticipate use of ImageMagick, MediaInfo and Tika, although specific technology choice requires further analysis.
```
Comment (ndushay) [inline 0001-technical-metadata-model.md]: can you add a clause of *what* you will do with these software packages?  auto-detect mime types, I assume?

**Comment by mjgiarlo** [comment]
Closing this PR as it's being superseded by https://github.com/sul-dlss-labs/infrastructure-adrs/pull/14

**Pull request closed without merging**